staff/ana-cano/precourse/bingo-pro/index.js

```diff
@@ -0,0 +1,175 @@
+// Definicion variables globales
+let turno = 0;
+let cantadoBingo = false;
+let cantandoLinea = false;
+let puntos = 100;
+let ranking = [];
+let numerosCantados=[];
+let numeroCantado = 0;
+let seguirJugando = true;
+let puntua = true;
+
+// TURNOS
+function generarCarton() {
+    return [
+        { number: getRandomNumber(1, 21), matched: false },
+        { number: getRandomNumber(20, 41), matched: false },
+        { number: getRandomNumber(40, 61), matched: false },
+        { number: getRandomNumber(60, 81), matched: false },
+        { number: getRandomNumber(80, 99), matched: false },
+        { number: getRandomNumber(1, 21), matched: false },
+        { number: getRandomNumber(20, 41), matched: false },
+        { number: getRandomNumber(40, 61), matched: false },
+        { number: getRandomNumber(60, 81), matched: false },
+        { number: getRandomNumber(80, 99), matched: false },
+        { number: getRandomNumber(1, 21), matched: false },
+        { number: getRandomNumber(20, 41), matched: false },
+        { number: getRandomNumber(40, 61), matched: false },
+        { number: getRandomNumber(60, 81), matched: false },
+        { number: getRandomNumber(80, 99), matched: false }
+    ]
+
+}
+
+// Cartón Bingo
+let carton = generarCarton();
+
+// FUNCIONES AYUDA
+
+//Números aleatorios cartón
+function getRandomNumber(min, max) {
+    return Math.round(Math.random() * (max - min) + min);
+}
+
+// Función Bombo
+function generarNumeroAleatorio(){
+    let numeroAleatorio = 0;
+    let numeroAleatorioOk = false;
+
+    numeroAleatorio = getRandomNumber(1,99);
+    
+    do {
+        // Comprobamos que numero aleatorio no este en numeros cantados
+        if (numerosCantados.indexOf(numeroAleatorio)==-1) {
+            numerosCantados.push(numeroAleatorio);
+            numeroAleatorioOk = true;
+        } else 
+        {
+            numeroAleatorio = getRandomNumber(1,99);
+        }
+    } while (numeroAleatorioOk === false);
+    
+    return numeroAleatorio;
+}
+
+
+//Función nuevo Turno
+function comprobarNumeroAleatorioEnCarton(param) {
+    for (let i = 0; i < carton.length; i++) {

```

Comment:
Hey! So just a performance tip. If you do "i < carton.length" javascript will evaluate the length every iteration. Why don't you try to create a variable outside of the loop with the value equal to the expression "carton.length". That way it will only evaluate that expression once and not every iteration. :)

Reply:
Hi I uploaded your recommendation . thank you